src/java/org/apache/cassandra/index/sai/utils/OrderRangeIterator.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.utils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+import org.apache.cassandra.io.util.FileUtils;
+
+// This is essentially a filter on top of a range iterator where we order chunks of primary keys,
```

Comment:
docstring

Reply:
... actually I'm not sure what's going on here, is this saying that it's doing order + limit repeatedly to avoid blowing up the resultset size?

Reply:
This iterator is meant to address this comment: https://github.com/riptano/VECTOR-SEARCH/issues/14#issuecomment-1729583877.

We read `x` PKs from the source iterator, fan out to each vector index, each index either orders and limits the PKs or choses to do a brute force comparison later, and then we get a stream of the remaining PKs.

Reply:
I will clean up the formatting

Reply:
Part of the problem is likely an incomplete abstraction. I need to look at the class again to see what will make the most sense.

Reply:
Fixed